app/service/rest_svc.py

```diff
@@ -40,8 +40,15 @@ async def persist_adversary(self, data):
             p = list()
             for ability in data.pop('atomic_ordering'):
                 p.append(ability['id'])
+            goals = list()
+            for goal in data.pop('goals'):
+                goal['count'] = int(goal['count'])
+                if goal['count'] < 0:
+                    goal['count'] = 2**20
+                goal['operator'] = goal['operator'].strip()
+                goals.append(goal)
```

Comment:
make this into a list comprehension that calls a private function to complete the operations